File: python-examples/14-router-coordinator/worker.py
"""
Polos Worker for the Blog Review example.

Run with:
    python worker.py
"""

import asyncio
import os

from dotenv import load_dotenv
from polos import PolosClient, Worker

from agents import (
    grammar_review_agent,
    tone_consistency_agent,
    correctness_agent,
    final_editor_agent,
    blog_generator_agent,
)
from workflows import blog_review, generate_blog

load_dotenv()


async def main():
    """Main function to run the worker."""
    project_id = os.getenv("POLOS_PROJECT_ID")
    if not project_id:
        raise ValueError(
            "POLOS_PROJECT_ID environment variable is required. "
            "Get it from the Polos UI at http://localhost:5173/projects/settings"
        )

    client = PolosClient(
        project_id=project_id,
        api_url=os.getenv("POLOS_API_URL", "http://localhost:8080"),
    )

    worker = Worker(
        client=client,
        workflows=[blog_review, generate_blog],
        agents=[
            grammar_review_agent,
            tone_consistency_agent,
            correctness_agent,
            final_editor_agent,
            blog_generator_agent,
        ],
    )

    print("Starting Blog Review Examples worker...")
    print(f"  Project ID: {project_id}")
    print(f"  Workflows: {[w.id for w in worker.workflows]}")
    print(f"  Agents: {[a.id for a in worker.agents]}")
    print("  Press Ctrl+C to stop\n")

    try:
        await worker.run()
    except KeyboardInterrupt:
        print("\nShutting down worker...")
        await worker.shutdown()


if __name__ == "__main__":
    asyncio.run(main())

File: typescript-examples/14-router-coordinator/worker.ts
/**
 * Polos Worker for the Blog Review example.
 *
 * Run with:
 *   npx tsx worker.ts
 *
 * Environment variables:
 *   POLOS_PROJECT_ID - Your project ID (required)
 *   POLOS_API_URL - Orchestrator URL (default: http://localhost:8080)
 *   POLOS_API_KEY - API key for authentication (optional for local development)
 *   POLOS_DEPLOYMENT_ID - Deployment ID (default: router-coordinator-examples)
 */

import 'dotenv/config';
import { Worker } from '@polos/sdk';
import {
  grammarReviewAgent,
  toneConsistencyAgent,
  correctnessAgent,
  finalEditorAgent,
  blogGeneratorAgent,
} from './agents.js';
import { blogReview, generateBlog } from './workflows.js';

async function main() {
  const projectId = process.env['POLOS_PROJECT_ID'];
  if (!projectId) {
    throw new Error(
      'POLOS_PROJECT_ID environment variable is required. ' +
        'Set it to your project ID (e.g., export POLOS_PROJECT_ID=my-project). ' +
        'You can get this from the output printed by `polos-server start` or from the UI page at ' +
        "http://localhost:5173/projects/settings (the ID will be below the project name 'default')",
    );
  }

  const apiUrl = process.env['POLOS_API_URL'] ?? 'http://localhost:8080';
  const apiKey = process.env['POLOS_API_KEY'] ?? '';
  const deploymentId = process.env['POLOS_DEPLOYMENT_ID'] ?? 'router-coordinator-examples';

  // In TS SDK, agents are also workflows — add both to the workflows array
  const allWorkflows = [
    blogReview,
    generateBlog,
    grammarReviewAgent,
    toneConsistencyAgent,
    correctnessAgent,
    finalEditorAgent,
    blogGeneratorAgent,
  ];

  const worker = new Worker({
    apiUrl,
    apiKey,
    projectId,
    deploymentId,
    workflows: allWorkflows,
  });

  console.log('Starting Blog Review Examples worker...');
  console.log(`  Project ID: ${projectId}`);
  console.log(`  Workflows: [${allWorkflows.map((w) => w.id).join(', ')}]`);
  console.log('  Press Ctrl+C to stop\n');

  await worker.run();
}

main().catch(console.error);
